docs/discussion/Multi-Tenancy Support.md

```diff
@@ -0,0 +1,168 @@
+Multi-Tenancy Support
+==
+Author: Binbin Li (@binbin-li)
+
+# Background
+Ratify’s first stable version for production has been officially released, garnering increased adoption as an external data provider among a growing number of customers. In production environments, numerous clients are opting for the implementation of a multi-tenant architecture to enhance resource efficiency within their clusters. To support this scenario, Ratify needs to make some essential refactoring. Enclosed in this document are several proposed design approaches and their respective modifications, warranting further scrutiny and discussion.
+
+# Multi-tenancy category
+Firstly, the concept of multi-tenancy within a Kubernetes cluster can be broadly categorized into two distinct classifications: multi-team and multi-customer.
+## Multiple teams
+In this case, cluster resources are shared between teams within the same organization. And it's usually scoped to a single cluster.
+## Multiple customers
+In this scenario, multi-tenancy support is usually provided by the service vendor. Customers do not have access to the cluster. And it may consume more resouces to support multiple customers.
+## Summary
+Given that Ratify primarily operates within a singular cluster, the chosen model for fostering multi-tenancy is the multi-team paradigm. In this framework, the responsibility for managing namespaces and allocating them to respective teams lies with the organization.
+
+# User Scenarios
+The overarching use case involves facilitating diverse cluster users in employing Ratify for the application of validation to target images using their individual configurations. However, to enhance the overall user experience, it is imperative to take into account several sub-scenarios.
+
+## User Group
+There could be several user groups/roles that have different access to the cluster.
```

Comment:
Are these three roles defined for Ratify or you assume these are general roles for the common case?

Reply:
those are general roles.